### PR TITLE
[CU-gh5cax] Get info of users from specific ms organization

### PR DESCRIPTION
**Pull request by eastemployeer**

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-gh5cax Get all users from specific MS organization](https://app.clickup.com/t/gh5cax)

File: server/database/repositories/TournamentRepository.ts [hunk 1]
```diff
@@ -17,20 +16,22 @@ export default class TournamentRepository {
...
-      if (field !== 'id') {
-        set[`matches.$.${field}`] = `${match[field as keyof Match]}`;
-      }
-    });
+    // const set: { [key: string]: string; } = {};
+    // Object.keys(match).forEach((field) => {
+    //   if (field !== 'id') {
+    //     set[`matches.$.${field}`] = `${match[field as keyof Match]}`;
+    //   }
+    // });
```
Comment (Zummek) [inline server/database/repositories/TournamentRepository.ts]: Maybe we should remove this?

File: server/routes/api-v1/organizationRouter.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import express from 'express';
-import * as authController from '../../controllers/authController';
+import * as MSController from '../../controllers/MSController';
 
 const organizationRouter = express.Router();
 
-organizationRouter.get('/logo', authController.getAzureADApplicationLogo);
+organizationRouter.get('/logo', MSController.getAzureADApplicationLogo);
+organizationRouter.get('/users', MSController.getAllUsers);
+organizationRouter.get('/users/:id/photo', MSController.getUserPhoto);
```
Comment (Zummek) [inline server/routes/api-v1/organizationRouter.ts:8]: We need some standard. I used to use singular words for naming resources, but i would like to change this in other resources to plural words, so i can add this as a task to backlog or change this in this branch

File: web-app/src/pages/Login.vue [hunk 3]
```diff
@@ -61,6 +61,8 @@ export default class Login extends Vue {
   private async mounted() {
     if (this.$route.query.code) {
       await api.auth.getCookieToken(this.$route.query.code as string);
+      // const response = await api.organization.getUsers();
+      // await api.organization.getUserPhoto(response[0].id);
```
Comment (Zummek) [inline web-app/src/pages/Login.vue:65]: ??? Here we should request for user details or better, get this data from token.

File: web-app/src/services/API/apiResources/organization.ts [hunk 4]
```diff
@@ -5,5 +5,24 @@ export const getAzureADApplicationLogo = async () => {
...
+export const getUsers = async () => {
+  const usersResponse = await axiosInstance.get<types.OrgUsersResponse>(
+    'v1/organization/users',
+    {
+      withCredentials: true,
+    }
+  );
+  return usersResponse.data.data.users;
+};
+export const getUserPhoto = async (id: string) => {
```
Comment (Zummek) [inline web-app/src/services/API/apiResources/organization.ts:20]: \n

File: web-app/src/services/API/apiResources/organization.ts [hunk 5]
```diff
@@ -5,5 +5,24 @@ export const getAzureADApplicationLogo = async () => {
...
+  const userPhotoResponse = await axiosInstance.get<types.OgUserPhotoResponse>(
+    `v1/organization/users/${id}/photo`,
```
Comment (Zummek) [inline web-app/src/services/API/apiResources/organization.ts:22]:
```suggestion
    `v1/organization/users/${encodeURIComponent(id)}/photo`,
```

File: web-app/src/services/API/apiResources/types.ts [hunk 6]
```diff
@@ -2,7 +2,32 @@ export interface AuthLoginResponse {
...
+  surname: string;
+  userPrincipalName: string;
+  id: string;
+}
+export interface OrgUsersResponse {
+  data: {
+    users: OrgUser[];
+  };
+}
+export interface OgUserPhotoResponse {
```
Comment (Zummek) [inline web-app/src/services/API/apiResources/types.ts:27]:
```suggestion
export interface OrgUserPhotoResponse {
```

File: web-app/src/services/API/apiResources/types.ts [hunk 7]
```diff
@@ -2,7 +2,32 @@ export interface AuthLoginResponse {
...
+  givenName: string;
+  jobTitle: string | null;
+  mail: string | null;
+  mobilePhone: string | null;
+  officeLocation: string | null;
+  preferredLanguage: string | null;
+  surname: string;
+  userPrincipalName: string;
+  id: string;
+}
```
Comment (Zummek) [inline web-app/src/services/API/apiResources/types.ts:21]: Most of this data is redundant

Comment (Zummek) [reply web-app/src/services/API/apiResources/types.ts:21]: Hmm `preferredLanguage` 
Would be nice to get this data from token about current user

Comment (eastemployeer) [reply web-app/src/services/API/apiResources/types.ts:21]: Maybe it is redundant but I wrote down whole object that comes with response from MS

File: web-app/src/services/API/apiResources/organization.ts (hunk 5 above)
Comment (eastemployeer) [reply web-app/src/services/API/apiResources/organization.ts:22]: What??? Why???

File: web-app/src/pages/Login.vue (hunk 3 above)
Comment (eastemployeer) [reply web-app/src/pages/Login.vue:65]: That was just for tests

File: server/database/repositories/TournamentRepository.ts (hunk 1 above)
Comment (eastemployeer) [reply server/database/repositories/TournamentRepository.ts]: We can

File: web-app/src/services/API/apiResources/types.ts (hunk 7 above)
Comment (Zummek) [reply web-app/src/services/API/apiResources/types.ts:21]: Got it, but we don't need a lot of properties when i just want a few. The rest of them can make programming harder

**Comment by Zummek** [comment]
@eastemployeer Is endpoint /users/:id/photo required? Will it be used for? 
In my opinion we should create /users/:id to get user object - we will be able to get user information and photo will be included in this object. Getting a photo of the user require getting the user object from the database (future) so it will not provide a problem to return the full object.

File: server/controllers/tournamentController.ts [hunk 8]
```diff
@@ -28,9 +28,14 @@ const updateMatchOutcomes = catchAsync(
 );
 
 const getAllTournaments = catchAsync(async (req, res) => {
-  const tournaments = await TournamentRepository.getAll();
+  const page = Number(req.query.page) || 1;
+  const pageSize = Number(req.query.pageSize) || 100;
+  const tournaments = await TournamentRepository.getAll(page, pageSize);
```
Comment (Zummek) [inline server/controllers/tournamentController.ts:33]: hmm what will happen when when i input `page: -50` or `pageSize: -100` or `page: 'XYZ'`? Do we need validation?

Comment (eastemployeer) [reply server/controllers/tournamentController.ts:33]: Mongoose will reject

**Pull request merged**